### PR TITLE
Add matcher for checking missing interpolation keys

### PR DESCRIPTION
**Pull request by creasty**
Wrote matcher for checking missing pairs of interpolation keys like:
`'I am %{name}'` and `'Soy %{nam3}'`
`'I am %{name}'` and `'Soy name'`

### CODE REVIEW

**Comment by aprescott** [comment]
This is a really nice addition. I think it would be good to actually include this as part of the functionality of the `be_a_subset_of` and `be_a_complete_translation_of` matchers. That way, not only are you asserting that translations aren't missing, you're also asserting that the interpolation keys didn't get altered or missed. What do you think?

**Comment by s01us** [comment]
I built something similar a while ago : https://github.com/tigrish/i18n-spec/pull/15

From experience I recommend a test case which passes as long as the keys are all present, regardless of order 

e.g expect
english: `%{name} is now an admin of %{group}`
bad_french:   `%{group}  a un nouvel admin %{name}`
to match

**Comment by creasty** [comment]
@aprescott

Well, actually I did in that way at the first place.
But I've changed it, after I thought translations didn't always have to fully contain interpolation keys across all languages.
And as I18.t, it simply ignores binding for missing keys, instead of raising an error.

So it's okay if `be_a_complete_translation_of` to take second argument to enable this feature.
How about that?

**Comment by aprescott** [comment]
@creasty interesting. I'm trying to imagine a case where you wouldn't want an interpolation key in one language even though it's used in another. I suppose if that's a legitimate use case, then yeah, an option argument seems fine to me! (Not a maintainer or owner of this repo, I just would really like to have the extra guarantee as a user of i18n-spec. :))

**Comment by creasty** [comment]
@s01us
It does, literally
https://github.com/tigrish/i18n-spec/raw/92c929732895270cd55ca552fa9fb99d3cc4dc9d/lib/i18n-spec/matchers/be_a_complete_interpolation_key_of_matcher.rb#L31

**Comment by creasty** [comment]
@aprescott aight, I'm gonna change it later

**Comment by creasty** [comment]
@s01us Ah, sorry I might have misunderstood your advise.
What you meant to say was to write a fixture having different order of keys so that spec could guarantee of its behavior.

**Comment by creasty** [comment]
@aprescott Added `check_interpolation_key` as the second argument to `be_a_subset_of` and `be_a_complete_translation_of`. And did little bit of refactoring.

File: lib/i18n-spec/matchers/be_a_complete_translation_of_matcher.rb [hunk 1]
```diff
@@ -1,14 +1,32 @@
-RSpec::Matchers.define :be_a_complete_translation_of do |default_locale_filepath|
+RSpec::Matchers.define :be_a_complete_translation_of do |default_locale_filepath, check_interpolation_key = false|
```
Comment (aprescott) [inline lib/i18n-spec/matchers/be_a_complete_translation_of_matcher.rb]: I don't know what the minimum version of Ruby is that is supported by this library, but it'd be great to use a keyword argument here of `check_interpolation_key: false`. I also have a feeling that a default value of `true` would be more useful.

File: lib/i18n-spec/interpolation_cheker.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        else
+          keys = extract_interpolation_keys t
+          result[scopes.join('.')] = keys unless keys.empty?
+        end
+
+        result
+      end
+
+      def extract_interpolation_keys(translation)
+        translation.to_s.scan(/(?<!%)%\{([^\}]+)\}/).flatten
```
Comment (aprescott) [inline lib/i18n-spec/interpolation_cheker.rb]: Is this a reimplementation of [I18n's existing `INTERPOLATION_PATTERN` constant](https://github.com/svenfuchs/i18n/blob/d4fb998394a389548e91d5f61af3ab77dd03472a/lib/i18n/interpolate/ruby.rb#L5-L9)?

```ruby
INTERPOLATION_PATTERN = Regexp.union(
  /%%/,
  /%\{(\w+)\}/,                               # matches placeholders like "%{foo}"
  /%<(\w+)>(.*?\d*\.?\d*[bBdiouxXeEfgGcps])/  # matches placeholders like "%<foo>.d"
)
```

Is it possible to reuse that here?

File: lib/i18n-spec/matchers/be_a_complete_translation_of_matcher.rb (hunk 1 above)
Comment (creasty) [inline lib/i18n-spec/matchers/be_a_complete_translation_of_matcher.rb]: Using keyword argument seems **to me** fine tho it's, in general, difficult problem to take Ruby version up to least 2.x unnecessarily.
And about the default value, I just want this feature to fail existing specs.

File: lib/i18n-spec/interpolation_cheker.rb (hunk 2 above)
Comment (creasty) [inline lib/i18n-spec/interpolation_cheker.rb]: It can't be replaced with that constant but can be copied from,

**Comment by creasty** [comment]
Quite uncertain whether @tigrish is alive...

**Comment by creasty** [comment]
I'm tired of waiting.
I've extracted this feature into my own gem without depending on `i18n-spec`:
https://github.com/creasty/i18n_interpolation_spec

**Comment by deivid-rodriguez** [comment]
@creasty Are you still interested in getting this merged in? I think it would be a valuable addition!

File: lib/i18n-spec/matchers/be_a_complete_translation_of_matcher.rb (hunk 1 above)
Comment (deivid-rodriguez) [reply lib/i18n-spec/matchers/be_a_complete_translation_of_matcher.rb]: I just dropped support for EOL'd rubies, so we can actually apply this suggestion now :+1:

**Pull request closed without merging**